vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -99,17 +99,43 @@
      */
     // <editor-fold desc="Configuration">
 
+    private static class LazyInitializationProvider {
+
+        @NotNull
+        private final Configuration mConfiguration;
+
+        public LazyInitializationProvider(@NotNull Configuration configuration) {
+            mConfiguration = configuration;
+        }
+
+        @NotNull
+        public Configuration getConfiguration() {
+            return mConfiguration;
+        }
+    }
+
+    private static LazyInitializationProvider sLazyInitializationProvider;
     private static VimeoClient mSharedInstance;
 
     public static VimeoClient getInstance() {
         if (mSharedInstance == null) {
+            if (sLazyInitializationProvider != null) {
+                initialize(sLazyInitializationProvider.getConfiguration());
+                if (mSharedInstance != null) {
+                    return mSharedInstance;
+                }
+            }
             throw new AssertionError("Instance must be configured before use");
         }
 
         return mSharedInstance;
     }
 
-    public static void initialize(Configuration configuration) {
+    public static void initializeLazily(@NotNull Configuration configuration) {
```

Comment:
This is exactly why I added all 3 of you to the ticket - should we just go with lazy loading over the current ```initialize``` approach? This does force all consumers to effectively lazy load until their first ```getInstance``` call, but is that a bad thing?